test/tests/functional/pbs_dup_acc_log_for_resv.py

```diff
@@ -47,16 +48,439 @@ class TestDupAccLogForResv(TestFunctional):
+            msg='.*B;' + rid, id=rid, n='ALL', allmatch=True, regexp=True)
+        self.assertNotEqual(m, None)
+        self.assertEqual(len(m), 1)
+
+        # Restart server
+        self.server.restart()
+
+        # Verify accounting log,no duplicate log after server restart
+
+        m = self.server.accounting_match(
+            msg='.*B;' + rid, id=rid, n='ALL', allmatch=True, regexp=True)
+        self.assertNotEqual(m, None)
+        self.assertEqual(len(m), 1)
+
+        a = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
+        self.server.expect(RESV, a, id=rid, max_attempts=30, offset=8)
+
+        # Restart server second time
+        self.server.restart()
+
+        # Verify accounting log,no duplicate log after server restart
+        m = self.server.accounting_match(
+            msg='.*B;' + rid, id=rid, n='ALL', allmatch=True, regexp=True)
+        self.assertNotEqual(m, None)
+        self.assertEqual(len(m), 1)
+
+    def test_standing_reservation(self):
+        """
+        Test for duplicate records in accounting log for standing reservations
+        on restart of server
+        """
+        # PBS_TZID environment variable must be set, there is no way to set
+        # it through the API call, use CLI instead for this test
+
+        _m = self.server.get_op_mode()
+        if _m != PTL_CLI:
+            self.server.set_op_mode(PTL_CLI)
+        if 'PBS_TZID' in self.conf:
+            tzone = self.conf['PBS_TZID']
+        elif 'PBS_TZID' in os.environ:
+            tzone = os.environ['PBS_TZID']
+        else:
+            self.logger.info('Missing timezone, using America/Los_Angeles')
+            tzone = 'America/Los_Angeles'
+        if _m == PTL_API:
+            self.server.set_op_mode(PTL_API)
+        instance1_start = int(time.time() + 5)
+        instance1_end = int(time.time() + 35)
+        a = {'Resource_List.select': '1:ncpus=1',
+             ATTR_resv_rrule: 'FREQ=MINUTELY;COUNT=2',
+             ATTR_resv_timezone: tzone,
+             'reserve_start': instance1_start,
+             'reserve_end': instance1_end}
+        r = Reservation(TEST_USER, a)
+        rid = self.server.submit(r)
+        a = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
+        self.server.expect(RESV, a, id=rid, offset=6)
+
+        ret = self.server.status(RESV, id=rid)
+        rid_instance1 = ret[0]['reserve_index']
+
+        # Verify accounting log before server restart
+        m = self.server.accounting_match(
+            msg='.*B;' + rid, id=rid, n='ALL', allmatch=True, regexp=True)
+        self.assertNotEqual(m, None)
+        self.assertEqual(len(m), 1)
+        # getting startime and endtime of first instance of standin reservation
+        got_instance1_start = self.check_log_resv_time(
+            accounting_log_msg=m, start_tup=0, end_tup=1, check_status="start")
+        got_instance1_end = self.check_log_resv_time(
+            accounting_log_msg=m, start_tup=0, end_tup=1, check_status="end")
```

Comment:
We can have check_log_resv_time() return multiple values and call it only once. 
Also, in all the calls check_log_resv_time(), end_tup is set to 1, only start_tup is changing, so we can remove this argument.